src/Jobs/PendingTransitionExecutor.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+
+namespace Asantibanez\LaravelEloquentStateMachines\Jobs;
+
+
+use Asantibanez\LaravelEloquentStateMachines\Exceptions\InvalidStartingStateException;
+use Asantibanez\LaravelEloquentStateMachines\Models\PendingTransition;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+
+class PendingTransitionExecutor implements ShouldQueue
+{
+    use InteractsWithQueue, Queueable, Dispatchable, SerializesModels;
+
+    public $pendingTransition;
+
+    public function __construct(PendingTransition $pendingTransition)
+    {
+        $this->pendingTransition = $pendingTransition;
+    }
+
+    /**
+     * @throws InvalidStartingStateException
+     */
+    public function handle()
+    {
+        $field = $this->pendingTransition->field;
+        $model = $this->pendingTransition->model;
+        $from = $this->pendingTransition->from;
+        $to = $this->pendingTransition->to;
+        $customProperties = $this->pendingTransition->custom_properties;
+
+        if ($model->$field()->isNot($from)) {
+            throw new InvalidStartingStateException();
```

Comment:
could also just `$this->fail()` here as it would be pointless to have retries after this. This would skip any retries that normally would occur.

Reply:
Might also be helpful to include why the state is invalid in the message of the exception.

Reply:
Just added `$this->fail()` and added a "expected/actual state" message to the exception. 👍